### ISSUE TITLE
A few questions about Knowledge Evolution

### ISSUE THREAD

**Issue by Booooooooooo** [ahmdtaha/knowledge_evolution#2]
Thank you for your exciting work! 
I've got a few questions after reading your paper, hoping you could help answer:
1. After g generations, do we need to finetune the fit hypothesis if we only want to use the fit hypothesis for inference?
2. Is the loss function a simple CE-loss for network N? It seems that you didn't put any explicit constrain to promote a zero-mapping inside the reset hypothesis. I could not quite understand why this will happen.

Thank you in advance~

**Comment by ahmdtaha** [comment]
Thanks for your interest in Knowledge evolution (KE)

1. Assuming we are talking about a single dataset, then no fine-tuning is required for the fit-hypothesis. In fact, I already extract the fit-hypothesis from the dense network N in my [code](https://github.com/ahmdtaha/knowledge_evolution/blob/7b81403d9613fe82008adda917ab4f0efd17198e/train_KE_cls.py#L82) after setting the [slim-factor](https://github.com/ahmdtaha/knowledge_evolution/blob/7b81403d9613fe82008adda917ab4f0efd17198e/train_KE_cls.py#L79). You can simply [save this model using pytorch](https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_models_for_inference.html), then load it later for inference. If you want to fine-tune, it is your decision. However, I would evaluate different split-rates.
2. Yes, just CE. Yes, I did not promote a zero-mapping inside the reset hypothesis **explicitly**. To understand how this works, I recommend [55], Section 3.3 (Intuition  `#2`), and **Appendix Intuition `#2`: Residual Network**. [55] shows that most residual links in a ResNet are close to zero. This means that a ResNet promotes a zero-mapping inside the residual links by having a **default** identity link. Similarly, KE promotes a zero-mapping inside the reset-hypothesis by having a **default** fit-hypothesis.

I hope my answers help. Thanks again for your interest in KE.


[55] Residual Networks Behave Like Ensembles of Relatively Shallow Networks

**Comment by Booooooooooo** [comment]
Thank you so much for kindly answering my questions. It helps a lot!
